# Boxscore Dataset

The boxscore dataset describes the performance of each team in the regular season.

- Each row represents a game between team1 (lower id) and team2 (higher id)
- For each boxscore statistic (e.g. ast - assists),
    - ast_team_mean1: average assists by team1
    - ast_team_std1: standard deviation of assists by team1
    - ast_opp_mean1: average assists by opponents of team1
    - ast_opp_std1: standard deviation of assists by opponents of team1

In [2]:
from src.data import make_dataset

In [18]:
data = make_dataset.get_boxscore_dataset_v1(season=2015)
data.head()

season  daynum  numot  tourney  team1  team2  score1  score2   loc  \
0    2015      11      0        0   1103   1420      74      57  1103   
1    2015      11      0        0   1104   1406      82      54  1104   
2    2015      11      0        0   1112   1291      78      55  1112   
3    2015      11      0        0   1113   1152      86      50  1113   
4    2015      11      0        0   1102   1119      78      84  1119   

   team1win  ... ast_opp_mean2  ast_opp_std2 to_opp_mean2  to_opp_std2  \
0         1  ...     13.600000      4.406891    12.766667     3.588135   
1         1  ...     11.258065      3.785655     9.290323     3.770000   
2         1  ...     11.400000      3.996550    13.700000     4.027064   
3         1  ...     12.896552      3.940131    14.137931     5.132001   
4         0  ...     14.068966      4.810190    13.206897     4.003385   

   stl_opp_mean2 stl_opp_std2  blk_opp_mean2  blk_opp_std2  pf_opp_mean2  \
0       8.700000     3.018906       3.066667      2.258063     18.166667   
1       6.129032     3.283717       4.516129      2.593209     20.064516   
2       6.333333     2.745948       2.733333      1.552158     16.800000   
3       7.310345     3.140785       3.517241      1.938716     15.758621   
4       7.000000     2.951997       3.620690      2.411487     18.793103   

   pf_opp_std2  
0     3.029548  
1     5.085505  
2     5.175539  
3     4.485335  
4     3.735728  

[5 rows x 128 columns]

## Example row
Here is an example of selected columns from the first row.

In [20]:
cols = ['team1', 'score1', 'score_team_mean1', 'score_team_std1', 'score_opp_mean1', 'score_opp_std1',
        'team2', 'score2', 'score_team_mean2', 'score_team_std2', 'score_opp_mean2', 'score_opp_std2',]
data.loc[[0], cols]

team1  score1  score_team_mean1  score_team_std1  score_opp_mean1  \
0   1103      74         67.352941         9.164762        63.235294   

   score_opp_std1  team2  score2  score_team_mean2  score_team_std2  \
0       11.214327   1420      57         56.066667         8.646041   

   score_opp_mean2  score_opp_std2  
0             67.1        8.539361

- team1 (1103) played against team2 (1420)
- team1 scored 74 and team2 scored 57
- average scoreline for team1 is 67.35 vs. 63.23. (win by 4 on average)
- average scoreline for team2 is 56.06 vs. 67.1   (lose by 11 on average)